benchexec/tools/ultimate.py

```diff
@@ -204,113 +186,120 @@ def _requires_ultimate_data(self, executable):
-                cmdline,
-                "cmdline contains empty or None argument when using default SVCOMP mode: ",
+        # there is no way to run ultimate; not enough parameters
+        msg = (
+            "Unsupported argument combination: You either need a property file or a toolchain option (-tc). "
+            "options={} property_file={} resource_limits={}".format(
+                options, task.property_file, resource_limits
             )
-            return cmdline
-
-        # if no property file is given and toolchain (-tc) is, use Ultimate directly
-        if "-tc" in options or "--toolchain" in options:
-            # ignore executable (old executable is just around for backwards compatibility)
-            mem_bytes = rlimits.get(MEMLIMIT, None)
-            cmdline = [self.java]
-
-            # -ea has to be given directly to java
-            if "-ea" in options:
-                options = [e for e in options if e != "-ea"]
-                cmdline += ["-ea"]
-
-            if mem_bytes:
-                cmdline += ["-Xmx" + str(mem_bytes)]
-            cmdline += ["-Xss4m"]
-            cmdline += ["-jar", self._get_current_launcher_jar(executable)]
-
-            if self._requires_ultimate_data(executable):
-                if "-ultimatedata" not in options and "-data" not in options:
-                    if self.api == 2:
-                        cmdline += [
-                            "-data",
-                            "@noDefault",
-                            "-ultimatedata",
-                            os.path.join(os.path.dirname(executable), "data"),
-                        ]
-                    if self.api == 1:
-                        raise ValueError(
-                            "Illegal option -ultimatedata for API {} and Ultimate version {}".format(
-                                self.api, self.version(executable)
-                            )
+        )
+        raise UnsupportedFeatureException(msg)
+
+    def _cmdline_no_wrapper(self, executable, options, task, resource_limits):
+        mem_bytes = resource_limits.memory
+        cmdline = [self.java]
+
+        # -ea has to be given directly to java
+        if "-ea" in options:
+            options = [e for e in options if e != "-ea"]
+            cmdline += ["-ea"]
+
+        if mem_bytes:
+            cmdline += ["-Xmx" + str(mem_bytes)]
+        cmdline += ["-Xss4m"]
+        cmdline += ["-jar", self._get_current_launcher_jar(executable)]
+
+        if self._requires_ultimate_data(executable):
+            if "-ultimatedata" not in options and "-data" not in options:
+                if self.api == 2:
+                    cmdline += [
+                        "-data",
+                        "@noDefault",
+                        "-ultimatedata",
+                        os.path.join(os.path.dirname(executable), "data"),
+                    ]
+                if self.api == 1:
+                    raise ValueError(
```

Comment:
Seems like `UnsupportedFeatureException` would be a good fit here and in similar locations?

Of course you could in principle also remove these checks and just let Ultimate print an error message when such a command line is attempted to be executed. It would be a little less convenient for the user, but reduce the maintenance effort of the tool-info module for you. So I leave this up to you.

Reply:
Yes, but I cannot change older Ultimate versions. So to be compatible with them we have a complex tool info module ;)